resim/metrics/default_report_metrics.py

```diff
@@ -0,0 +1,729 @@
+    batch_to_metrics_map: dict[str, list[BatchMetric]],
+    scalar_batch_metrics_map: dict[str, dict[str, MetricProto]],
+) -> None:
+    # pylint: disable=unused-argument
+    """Create a variety of job status metrics.
+
+    Includes a filled line plot of statuses alongside aggregate metrics on:
+     - Number of batches
+     - Number of jobs
+     - Number of passing batches
+     - Number of failing batches
+     - Failed jobs on most recent batch
+     - etc.
+
+    Args:
+        writer: A metrics writer to write these metrics to.
+        batches: All batches for this report.
+        batch_to_jobs_map: All jobs for each batch.
+        job_to_metrics_map: All job metrics for each job.
+        batch_to_metrics_map: All batch metrics for each batch.
+        scalar_batch_metrics_map: All scalar batch metrics for each batch as protos.
+    """
+    status_counts = _count_batch_statuses(
+        list(batch_to_jobs_map.keys()), batch_to_jobs_map
+    )
+
+    fig = px.area(
+        status_counts,
+        x=range(len(batch_to_jobs_map)),
+        y=["PASSED", "FAIL_WARN", "FAIL_BLOCK", "ERROR", "CANCELLED", "UNKNOWN"],
+    )
+    fig.update_layout(
+        template="plotly_dark",
+        plot_bgcolor="rgba(0, 0, 0, 0)",
+        paper_bgcolor="rgba(0, 0, 0, 0)",
+    )
+
+    (
+        writer.add_plotly_metric("Job Statuses Over Time")
+        .with_description("Job Statuses Over Sequential Batches")
+        .with_blocking(False)
+        .with_should_display(True)
+        .with_importance(mu.MetricImportance.HIGH_IMPORTANCE)
+        .with_status(mu.MetricStatus.PASSED_METRIC_STATUS)
+        .with_plotly_data(fig.to_json())
+    )
+
+    fail_statuses = (MetricStatus.FAIL_BLOCK, MetricStatus.FAIL_WARN)
+
+    def batch_is_failure(batch: Batch) -> bool:
+        if batch.status == BatchStatus.ERROR:
+            return True
+
+        if batch.status != BatchStatus.SUCCEEDED:
+            return False
+
+        return (
+            batch.batch_metrics_status in fail_statuses
+            or batch.jobs_metrics_status in fail_statuses
+        )
+
+    def batch_is_success(batch: Batch) -> bool:
+        if batch.status != BatchStatus.SUCCEEDED:
+            return False
+
+        return (
+            batch.batch_metrics_status not in fail_statuses
+            and batch.jobs_metrics_status not in fail_statuses
+        )
+
+    if len(batches) == 0:
+        raise ValueError("Can't compute totals on zero batches!")
+
+    batches_sorted = all(
+        batches[i].creation_timestamp < batches[i + 1].creation_timestamp
+        for i in range(len(batches) - 1)
+    )
+
+    if not batches_sorted:
+        raise ValueError("Batches must be sorted!")
+
+    totals = {
+        "Number of Batches": len(batches),
+        "Number of Jobs": len(job_to_metrics_map),
+        "Number of Passing Batches": sum(batch_is_success(b) for b in batches),
+        "Number of Failing Batches": sum(batch_is_failure(b) for b in batches),
```

Comment:
hmm. I think that fail isn't the right word here. I think it might be better to separate out the three types?

Or, the web app uses the term (which i don't particularly like) "Issues"

Reply:
(below uses a different definition of fail)

Reply:
I think I need some guidance on how to break these out. For instance, how do we want to categorize a situation whose batch metrics are fail warn but whose job metrics are fail block? I guess we should pick the more severe option?

Reply:
Discussed offline and decided to be conservative (select more severe options).